components/esp_eth/src/esp_eth_phy_ksz80xx.c

```diff
@@ -49,6 +50,26 @@ typedef union {
 } ksz8041_pc2r_reg_t;
 #define KSZ8041_PC2R_REG_ADDR (0x1F)
 
+/**
+ * @brief PC1R(PHY Control 1 Register) for KSZ8061
+ *
+ */
+typedef union {
+    struct {
+        uint32_t op_mode : 3;         /* Operation Mode Indication */
+        uint32_t phy_iso : 1;         /* PHY in Isolate Mode*/
+        uint32_t energy_det : 1;      /* Presence of Signal on RX+/- Wire Pair */
+        uint32_t mdix_state : 1;      /* MDI/MDI-X State */
+        uint32_t reserved_6 : 1;      /* Reserved */
+        uint32_t polarity_status : 1; /* Polarity Status */
+        uint32_t lnk_status : 1;      /* Link Status */
+        uint32_t flow_ctl_cap : 1;    /* Flow Control Capable */
+        uint32_t reserved_15_10: 6;   /* Reserved */
+    };
+    uint32_t val;
+} ksz8061_pc1r_reg_t;
+#define KSZ8061_PC1R_REG_ADDR (0x1E)
```

Comment:
This register seems to have exactly the same layout as KSZ8081 one. Please double check what are the differences between KSZ8061 and KSZ8081. It could help us to simplify the driver's implementation.

Reply:
Well, they are different chips with different features. KSZ8061 is recommended for automotive designs and has extended temperature range compared with KSZ8081. Although they share the same PHY Control 1 and (almost) PHY Control 2 registers, they have different registers for further purposes.

So I do not recommend to set them as equal ([as done for the KSZ8091 and KSZ8081](https://github.com/espressif/esp-idf/blob/7af719e426952995b2a205e821e7471a0b995125/components/esp_eth/include/esp_eth_phy.h#L309-L312)). It may work with this implementation as only the PC1R register is used. But as a developer, when modifiing/extending source code for example at [these lines](https://github.com/espressif/esp-idf/blob/7af719e426952995b2a205e821e7471a0b995125/components/esp_eth/src/esp_eth_phy_ksz80xx.c#L131-L135), I would only check the KSZ8081 datasheet and would not expect that it is also executed for the KSZ8091 (that is in fact)...

Reply:
Well, the first question we need to ask is "What is a purpose of this driver". From my perspective, it is support of basic functionality of KSZ80xx family. If the basic functionality has common control interface, we should keep it common too. If each chip provides some further features, these need to be handled separately, of course. However, this is probably not what should be supported directly by Espressif. I would focus on basic functionality and let users to extend the driver for their specific needs by themselves and maintain it using [Component Manager](https://components.espressif.com/search/target:esp32) if they wanted to share their work with community.  "Extended temperature range" does not make any difference from SW point of view, in my opinion.

Speaking of KSZ8091 and KSZ8081. The basic functionality is the same. They even have the same Vendor ID (quite surprisingly for me). If you check how the KSZ8091 is initialized (calling KSZ8081 function) and if you consider that they have the same Vendor ID, then I do not see any significant problem with referenced lines.

Anyway, we really need to think about it deeper to make it right and understandable... I am currently busy with other tasks but I am open to suggestions.

Reply:
Inspired by a [phy implementation in the android open source project](https://github.com/aosp-mirror/kernel_common/blob/dc7512451c7a83cb8a02746afac4f5bca2109c47/drivers/net/phy/micrel.c#L126-L147), I moved to dynamically specifying the op_mode register and offset in the `phy_ksz80xx_t` type. This totally resolves the issue with the different (or equal) register type definitions. They are not needed anymore. I think the way of accessing register information based on phy model in this way is pretty future proof and easily extensible.